### ISSUE TITLE
Feature: Use current block - 3 reference for broadcast tx

### ISSUE THREAD

**Issue by Netherdrake** [steemit/steem-python#13]
todo: https://github.com/steemit/steem-js/pull/161/files

**Comment by bitfag** [comment]
Guys could you explain why this change was introduced? I'm often getting an errors while making transactions. I'm on golos blockchain and I'm using a forked version of steem-python.

Here is the my report to golos-python:
https://github.com/GolosChain/golos-python/issues/3

I reverted this change and errors disappeared.

**Comment by GarethNelson** [comment]
This might explain a recent bug in yo too

**Comment by cyon1c** [comment]
@sneak 
@john-g-g 

Is this issue still relevant?

[steem-python does not currently implement this.](https://github.com/steemit/steem-python/blob/1d7e7f3a62ec180a986f0d4504939e822bc194be/steem/transactionbuilder.py#L89)

**Comment by sneak** [comment]
Yes, it is likely still relevant.  If the change is introducing problems then it's likely a bug in the change, not the concept.

Each tx needs to refer to the block hash of a recent block.  A block from a few blocks ago should be used, instead of the absolute tip of the chain, in the event of a microfork that invalidates the tip block which then renders the transactions that referenced it invalid.

**Comment by bitfag** [comment]
@sneak I think reasonable default is use last_irreversible_block as a reference. This will prevent transactions loosing at chain forks.
